x-pack/plugins/security_solution/public/common/containers/use_full_screen/index.tsx

```diff
@@ -85,3 +107,14 @@ export const useTimelineFullScreen = (): TimelineFullScreen => {
   );
   return memoizedReturn;
 };
+
+export const useIsDataGridExpanded = () => {
+  const [isDataGridExpanded, setDataGridExpanded] = useState(false);
+  const onMutation = useCallback(() => {
+    const isExpanded = document.querySelector('.euiDataGrid--fullScreen') !== null;
+    setDataGridExpanded(isExpanded);
+  }, [setDataGridExpanded]);
```

Comment:
i think you should be able to not have this setter as a dependency because it is coming from useState